File: 2020 kwiecien/python/3.py
with open("dane4.txt", "r") as f:
    liczby = list(map(int, f.read().strip().split('\n')))

cnt = dict()
for i in range(1, len(liczby)):
    if abs(liczby[i] - liczby[i - 1]) not in cnt.keys():
        cnt[abs(liczby[i] - liczby[i - 1])] = 0
    cnt[abs(liczby[i] - liczby[i - 1])] += 1

best = max(cnt.values())
print("Krotność najczęstszej luki:", best)
print("Wartości najczęstszych luk:")
for i, j in cnt.items():
    if j == best:
        print(i)

File: 2020 kwiecien/c++/3.cpp
#include <iostream>
#include <fstream>
#include <map>
#include <algorithm>
using namespace std;

int a[1005];
map <int, int> m;

int main()
{
    ifstream in("dane4.txt");

    for (int i = 1; i <= 1000; i++)
        in >> a[i];
    
    for (int i = 2; i <= 1000; i++)
        m[abs(a[i] - a[i - 1])]++;
    
    int best = 0;
    for (pair <int, int> x: m)
        best = max(best, x.second);

    cout << "Krotnosc najczestszej luki: " << best << '\n';    
    cout << "Wartosci najczestszych luk:\n";
    for (pair <int, int> x: m)
        if (x.second == best) cout << x.first << '\n';
}